src/integrators/controllers.jl

```diff
@@ -127,13 +127,13 @@ end
   @unpack qold = integrator
   @unpack qmin, qmax, gamma = integrator.opts
   @unpack beta1, beta2 = controller
-  EEst = integrator.EEst
+  EEst = DiffEqBase.value(integrator.EEst)
```

Comment:
Can you double check this works on the other controllers too?

Reply:
The `IController` and the `PIController` are fine. The tests for
```
g5 = Zygote.gradient(θ->test_f2(θ,ReverseDiffAdjoint(), PIDController(1/18. , 1/9., 1/18.)), p)
```
errors:
```julia
julia> Zygote.gradient(θ->test_f2(θ,ReverseDiffAdjoint(), PIDController(1/18. , 1/9., 1/18.)), p)
integrator.t = TrackedReal<GFL>(0.4517539514526256, 0.0, KaQ, ---)
bounced.
ERROR: BoundsError: attempt to access 0-element Vector{Any} at index []
Stacktrace:
  [1] throw_boundserror(A::Vector{Any}, I::Tuple{})
    @ Base ./abstractarray.jl:651
  [2] checkbounds
    @ ./abstractarray.jl:616 [inlined]
  [3] _getindex
    @ ./abstractarray.jl:1196 [inlined]
  [4] getindex
    @ ./abstractarray.jl:1170 [inlined]
  [5] macro expansion
    @ ~/.julia/packages/Zygote/i1R8y/src/lib/lib.jl:298 [inlined]
  [6] (::Zygote.Jnew{MVector{3, Float64}, Vector{Any}, false})(Δ::Nothing)
    @ Zygote ~/.julia/packages/Zygote/i1R8y/src/lib/lib.jl:292
  [7] #1723#back
    @ ~/.julia/packages/ZygoteRules/OjfTt/src/adjoint.jl:59 [inlined]
  [8] Pullback
    @ ~/.julia/packages/StaticArrays/ECfU2/src/MArray.jl:30 [inlined]
  [9] (::typeof(∂(MVector{3, Float64})))(Δ::Nothing)
    @ Zygote ~/.julia/packages/Zygote/i1R8y/src/compiler/interface2.jl:0
 [10] Pullback
    @ ~/.julia/packages/StaticArrays/ECfU2/src/convert.jl:4 [inlined]
 [11] Pullback
    @ ~/.julia/dev/OrdinaryDiffEq/src/integrators/controllers.jl:241 [inlined]
 [12] (::typeof(∂(#PIDController#511)))(Δ::Nothing)
    @ Zygote ~/.julia/packages/Zygote/i1R8y/src/compiler/interface2.jl:0
 [13] Pullback
    @ ~/.julia/dev/OrdinaryDiffEq/src/integrators/controllers.jl:239 [inlined]
 [14] (::typeof(∂(PIDController)))(Δ::Nothing)
    @ Zygote ~/.julia/packages/Zygote/i1R8y/src/compiler/interface2.jl:0
 [15] Pullback
    @ ./none:1 [inlined]
 [16] (::typeof(∂(#81)))(Δ::Float64)
    @ Zygote ~/.julia/packages/Zygote/i1R8y/src/compiler/interface2.jl:0
 [17] (::Zygote.var"#41#42"{typeof(∂(#81))})(Δ::Float64)
    @ Zygote ~/.julia/packages/Zygote/i1R8y/src/compiler/interface.jl:41
 [18] gradient(f::Function, args::Vector{Float64})
    @ Zygote ~/.julia/packages/Zygote/i1R8y/src/compiler/interface.jl:59
 [19] top-level scope
    @ none:1
```
I don't imediately see why it doens't work though https://github.com/SciML/OrdinaryDiffEq.jl/blob/e50ebf5c2c7b9ee8bdf079919add78851796d8f5/src/integrators/controllers.jl#L241.

Reply:
the forward solution works:
```julia
solve(prob,Tsit5(),controller=PIDController(1/18. , 1/9., 1/18.),
  abstol=1e-14,reltol=1e-14,callback=cb,save_everystep=false)
```
However, both 
```
Zygote.gradient(θ->test_f2(θ,ReverseDiffAdjoint(), PIDController(1/18. , 1/9., 1/18.)), p)
Zygote.gradient(θ->test_f2(θ,ForwardDiffSensitivity(), PIDController(1/18. , 1/9., 1/18.)), p)
```
fail.